src/script/microbit-interfacing/MicrobitUSB.ts

```diff
@@ -122,8 +125,35 @@ class MicrobitUSB extends CortexM {
     return Promise.resolve();
   }
 
-  public get serialPort(): SerialPort {
-    return new WebUSBSerialDevice(this.transport);
+  public async startSerial(callback: (data: string) => void) {
+    await this.transport.open();
+
+    this.serialDAPLink = new DAPLink(this.transport);
+    const initialBaudRate = await this.serialDAPLink.getSerialBaudrate();
+    this.serialDAPLink.addListener(DAPLink.EVENT_SERIAL_DATA, callback);
+    await this.serialDAPLink.connect();
+    if (initialBaudRate !== baudRate) {
+      await this.serialDAPLink.setSerialBaudrate(baudRate);
+    }
+    this.serialPromise = this.serialDAPLink.startSerialRead(5, false);
```

Comment:
Is this configuring the pulling rate to 5ms? Maybe we should set that as a local const like baudRate?

Reply:
Yeah sure I'll pull it out. 

Value TBD as we test, but I figure it should be more > 2x more frequent than the writes to get timely data.

We'll follow up with some code to log periodic bucketed performance stats as the console.logs seem to have something of an effect on performance.

Reply:
38a1cb4e4e847bbd8f5b826f619b739c7083a518